Source/core/MessageDispatcher.h

```diff
@@ -0,0 +1,415 @@
+         */
+        uint32_t PushMetadata(const uint16_t length, const uint8_t* value)
+        {
+            _metaDataLock.Lock();
+
+            ASSERT(length > 0);
+
+            uint32_t result = Core::ERROR_GENERAL;
+
+            Core::IPCChannelClientType<Core::Void, false, true> channel(Core::NodeId(_filenames.metaData.c_str()), METADATA_SIZE);
+
+            auto metaDataFrame = Core::ProxyType<typename MetaDataBuffer<METADATA_SIZE>::MetaDataFrame>::Create();
+
+            if (channel.Open(Core::infinite) == Core::ERROR_NONE) {
+                metaDataFrame->Parameters().Set(length, value);
+
+                if (channel.Invoke(metaDataFrame, Core::infinite) == Core::ERROR_NONE) {
+                    result = metaDataFrame->Response();
+                } else {
+                    result = Core::ERROR_WRITE_ERROR;
+                }
+
+                channel.Close(Core::infinite);
+            }
+
+            _metaDataLock.Unlock();
+
+            return result;
+        }
+
+        void RegisterDataAvailable(MetaDataCallback notification)
+        {
+            if (_metaDataBuffer->IsOpen()) {
+                _metaDataBuffer->RegisterMetaDataCallback(notification);
+            }
+        }
+        void UnregisterDataAvailable()
+        {
+            _metaDataBuffer->UnregisterMetaDataCallback();
+        }
+
+        bool IsValid()
+        {
+
+            bool result = true;
+
+            if (!_dataBuffer.IsValid()) {
+                result = _dataBuffer.Validate();
+            }
+            if (_metaDataBuffer != nullptr) {
+                if (!_metaDataBuffer->IsOpen()) {
+                    result = false;
+                }
+            }
+            return result;
+        }
+
+    private:
+        struct Filenames {
+            string doorBell;
+            string metaData;
+            string data;
+        } _filenames;
+
+        /**
+        * @brief Prepare filenames for MessageDispatcher
+        * 
+        * @param initialize should the base directory be created (if it exists everythiing inside will be deleted)
+        * @param baseDirectory where are those filed stored
+        * @param identifier identifer of the instance
+        * @param instanceId number of instance
+        * @return std::tuple<string, string, string> 
+        *         0 - doorBellFilename
+        *         1 - dataFileName
+        *         2 - metaDataFilename
+        */
+        static Filenames PrepareFilenames(const bool initialize, const string& baseDirectory, const string& identifier, const uint32_t instanceId)
+        {
+            if (initialize) {
+                if (Core::File(baseDirectory).IsDirectory()) {
+                    //if directory exists remove it to clear data (eg. sockets) that can remain after previous creation
+                    Core::Directory(baseDirectory.c_str()).Destroy(false);
+                }
+
+                if (!Core::Directory(baseDirectory.c_str()).CreatePath()) {
```

Comment:
Same for creating the Directory, leave this up to the Higher Layer. Suggest to ASSERT here if the Directory does not exist :-)